gdal/frmts/nitf/nitfimage.c

```diff
@@ -657,6 +657,16 @@ NITFImage *NITFImageAccess( NITFFile *psFile, int iSegment )
                     psImage->nTREBytes );
 
             nOffset += nUserTREBytes;
+
+            sscanf(psImage->pachTRE + 6, "%*5d%n", &nFirstTagUsedLength);
+            if (nFirstTagUsedLength != 5)
+            {
+                CPLError(CE_Failure, CPLE_AppDefined,
```

Comment:
as the processing of the image is not completely stopped, it would be better to use a CE_Warning

Reply:
Changed.